userspace/libsinsp/parsers.cpp

```diff
@@ -1097,15 +1097,15 @@ void sinsp_parser::parse_clone_exit_caller(sinsp_evt *evt, int64_t child_tid)
 	/*=============================== CHILD ALREADY THERE ===========================*/
 
 	/* See if the child is already there, if yes and it is valid we return immediately */
-	sinsp_threadinfo* child_tinfo = m_inspector->get_thread_ref(child_tid, false, true).get();
-	if(child_tinfo != nullptr)
+	sinsp_threadinfo* existing_child_tinfo = m_inspector->get_thread_ref(child_tid, false, true).get();
```

Comment:
I assume we keep the raw pointer here because we don't want the overhead of creating a `std::shared_ptr` here, if so, why is it ok to create a `std::unique_ptr` in line 1128? That also has a potential overhead.

Reply:
I saw your previous comment on `std::unique_ptr` not being as expensive as `std::shared_ptr`, I agree with it, but this is still the hot path for process creation, I'd argue any potential performance hit needs to at least be mentioned 🤷🏻 .

Reply:
The reason relies both in ownership and performance. In this case, operating on a copy of the shared pointer would definitely add performance burden due to reference counting, which is totally useless given that we use it inline only for this portion of code and then return. From an ownership perspective, it also does not make sense to allocate a shared pointer because the function never _owns_ the pointer -- it just uses it and then return. In the Rust world, this would be a reference borrow, and here could be definitely a C++ reference. However, given that the returned thread can legitimately be null, using C++ reference does not make much sense either, hence why this is preserved as a raw pointer for now. IMO, this is one of the cases in which a raw pointer makes sense without adding safety concerns.